tests/Integration/Behaviour/Features/Context/Domain/ApiAccessManagementFeatureContext.php

```diff
@@ -0,0 +1,172 @@
+ */
+
+declare(strict_types=1);
+
+namespace Tests\Integration\Behaviour\Features\Context\Domain;
+
+use Behat\Gherkin\Node\TableNode;
+use PrestaShop\PrestaShop\Core\Domain\ApiAccess\Command\AddApiAccessCommand;
+use PrestaShop\PrestaShop\Core\Domain\ApiAccess\Command\EditApiAccessCommand;
+use PrestaShop\PrestaShop\Core\Domain\ApiAccess\Exception\ApiAccessConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\ApiAccess\Query\GetApiAccessForEditing;
+use PrestaShop\PrestaShop\Core\Domain\ApiAccess\ValueObject\ApiAccessId;
+use Tests\Integration\Behaviour\Features\Context\CommonFeatureContext;
+use Tests\Integration\Behaviour\Features\Context\Util\DataComparator;
+use Tests\Integration\Behaviour\Features\Context\Util\DataTransfer;
+use Tests\Resources\Resetter\ApiAccessResetter;
+
+class ApiAccessManagementFeatureContext extends AbstractDomainFeatureContext
+{
+    /**
+     * @When /^I create an api access "(.+)" with following properties:$/
+     */
+    public function createApiAccessUsingCommand($apiAccessReference, TableNode $table)
+    {
+        $data = $this->fixDataType($table->getRowsHash());
+        $commandBus = $this->getCommandBus();
+
+        $command = new AddApiAccessCommand(
+          $data['clientName'],
+          $data['apiClientId'],
+          $data['enabled'],
+          $data['description']
+        );
+
+        try {
+            /** @var ApiAccessId $id */
+            $id = $commandBus->handle($command);
+
+            $this->getSharedStorage()->set($apiAccessReference, $id->getValue());
+        } catch (ApiAccessConstraintException $e) {
+            $this->setLastException($e);
+        }
+    }
+
+    /**
+     * @Then /^api access "(.+)" should have the following properties:$/
+     */
+    public function assertQueryApiAccessProperties($apiAccessReference, TableNode $table)
+    {
+        $expectedData = $table->getRowsHash();
+        $this->getSharedStorage()->exists($apiAccessReference);
+
+        $queryBus = $this->getQueryBus();
+        $result = $queryBus->handle(new GetApiAccessForEditing($this->getSharedStorage()->get($apiAccessReference)));
+
+        $serializer = CommonFeatureContext::getContainer()->get('serializer');
+        $realData = $serializer->normalize($result);
+
+        DataComparator::assertDataSetsAreIdentical($expectedData, $realData);
+    }
+
+    /**
+     * @When /^I edit api access "(.+)" with the following values:$/
+     */
+    public function editCustomerUsingCommand($apiAccessReference, TableNode $table)
+    {
+        $this->getSharedStorage()->exists($apiAccessReference);
+        $data = $table->getRowsHash();
+
+        $commandBus = $this->getCommandBus();
+
+        $command = new EditApiAccessCommand($this->getSharedStorage()->get($apiAccessReference));
+        $data = $this->fixDataType($data);
+
+        DataTransfer::transferAttributesFromArrayToObject($data, $command);
```

Comment:
Here also not a fan because it hides the usage of getters/setters and it forces using a convention to match the fields, if someone changes a key in the table he probably won't understand why it doesn't work anymore

Another advantage of using the getters/setters manually is that when you do refacto you can actually see all the usages even in tests

But again personal opinion, not blocking as long as the tests check what they're supposed to